src/navigation/drive.py

```diff
@@ -1,76 +1,118 @@
 from __future__ import annotations
-from typing import Tuple
-
-
+from typing import Tuple, Optional
 import numpy as np
 import rospy
+from enum import Enum
 
 from geometry_msgs.msg import Twist
 from util.SE3 import SE3
 from util.np_utils import angle_to_rotate
 from util.ros_utils import get_rosparam
 
+default_constants = {
+    "max_driving_effort": 1.0,
+    "min_driving_effort": -1.0,
+    "turning_p": 10.0,
+    "max_turning_effort": 1.0,
+    "min_turning_effort": -1.0,
+    "driving_p": 1.0,
+}
+ODOM_CONSTANTS = get_rosparam("drive/odom", default_constants)
+MAP_CONSTANTS = get_rosparam("drive/map", default_constants)
+
+
+class Driver:
+    _last_angular_error: Optional[float] = None
+
+    class DriveMode(Enum):
+        TURN_IN_PLACE = 1
+        DRIVE_FORWARD = 2
+        STOPPED = 3
+
+    _driver_state: DriveMode = DriveMode.STOPPED
+
+    def _is_complete(self, target_pos: np.ndarray, rover_pose: SE3, completion_thresh: float) -> bool:
+        rover_pos = rover_pose.position
+        target_dir = target_pos - rover_pos
+        target_dist = np.linalg.norm(target_dir)
+        return bool(target_dist < completion_thresh)
```

Comment:
mypy was complaining: ` error: Incompatible return value type (got "bool_", expected "bool") `

Reply:
Try type hinting target_dist as float

Reply:
this is not relevant anymore but I had to do something similar on line 151. Just type hinting it didn't work I had to cast. The error without that cast is: `incompatible types in assignment (expression has type "floating[Any]", variable has type "float")` I think this may be a numpy issue not sure if its fixed in newer versions